src/server.c

```diff
@@ -482,8 +485,20 @@ static void decorate_connection(qd_server_t *qd_server, pn_connection_t *conn, c
         pn_data_put_int(pn_connection_properties(conn), config->inter_router_cost);
     }
 
-    if (config) {
+    if (ctx->connector && ctx->connector->is_data_connector) {
+        pn_data_put_symbol(pn_connection_properties(conn),
+                           pn_bytes(strlen(QD_CONNECTION_PROPERTY_ROLE_KEY), QD_CONNECTION_PROPERTY_ROLE_KEY));
+        pn_data_put_int(pn_connection_properties(conn), QDR_ROLE_INTER_ROUTER_DATA);
+    }
+
+    if (ctx->connector && (ctx->connector->is_data_connector || ctx->connector->config.has_data_connectors)) {
+        pn_data_put_symbol(pn_connection_properties(conn),
+                           pn_bytes(strlen(QD_CONNECTION_PROPERTY_GROUP_CORRELATOR_KEY), QD_CONNECTION_PROPERTY_GROUP_CORRELATOR_KEY));
+        pn_data_put_string(pn_connection_properties(conn),
```

Comment:
Looking at the protocol trace this appears to include the null termination byte when written to the wire:]

[C2]:FRAME: 0 -> @open(16) [container-id="TestRouterB", ... :"qd.conn-id"=2, :"qd.inter-router-role"=4, :"qd.group-correlator"="PJ2n9qvOMWiyG2y\x00"}]

QD_DISCRIMINATOR_SIZE is for sizing buffers to hold the discriminator and the trailing null octet.  Maybe just use strlen?